### PR TITLE
Updates placeholder for post comment block

### PR DESCRIPTION
**Pull request by karmatosed**
This is a little PR that cleans up the placeholder on post comment block. Whilst a lot of these content type blocks are a work in progress, there are some visual hitches that can be smoothed. Primarily what this does:

- Brings the input inline with height of other inputs (for example table placeholder).
- Adds space between the input and save (as per other blocks).
- Adjusts copy to be a little clearer what it does.

**Before:**

image

**After:**

after-postcomments

## Feedback
I specifically would like a code review, but this shouldn't need much feedback as really just aligning with other blocks. Of course, any input is always welcome.

### CODE REVIEW

**Comment by karmatosed** [comment]
Looping in @jameskoster here as potentially this block will get iterated outside of this PR. My thinking is this visual fix can ease whilst that's getting done, but happy to close now I know about #26856.

File: packages/block-library/src/post-comments/editor.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.wp-block-post-comment {
+
+    .components-base-control {
+        margin-right: $grid-unit-10;
+    }
+ 
+    input {
+        height: 36px;
```
Comment (jasmussen) [inline packages/block-library/src/post-comments/editor.scss:8]: You can replace 36px with `$button-size` here.

**Comment by jasmussen** [comment]
Cool PR.

I think it needs a slightly different approach though. For a basic placeholder such as this one, you shouldn't need to create a new editor style just to fix this layout issue, those should be fixed at the component level, i.e. so every future placeholder like this works.

So I did a little inspecting, you can look here:

change

So it looks from the above that `.components-placeholder__input[type=url]` has the right CSS rules to make this look good, but that the input field used for the Post Comment placeholder, for some reason, does not inherit those rules. So either it's missing a classname (probably missing the `components-placeholder__input` class) or it's something else.

**Comment by karmatosed** [comment]
@jasmussen Thanks for the review. I did remove the styles and agree there is something deeper going on here. I did some digging and could manage to get the class to wrap everything, but not on specific element. I'd happily learnt that though in this case.

**Comment by jasmussen** [comment]
Yep, looks like this uses the `<TextControl>` component, which is not aware of its placeholder context and therefore doesn't inherit the same rules as a plain old `<input>` field with a `components-placeholder__input` class attached. So it seems that needs a separate look-see. @ItsJonQ I wouldn't ask you to work on this — only share your thoughts on what the best way to systematize this is. 

In the mean time, as a description change, we can probably just get this in.

**Review by jasmussen — APPROVED** [review]
It doesn't fix the clunky input spacing, but it fixes the description.

**Comment by karmatosed** [comment]
> Yep, looks like this uses the <TextControl> component, which is not aware of its placeholder context and therefore doesn't inherit the same rules as a plain old <input> field with a components-placeholder__input class attached. So it seems that needs a separate look-see. @ItsJonQ I wouldn't ask you to work on this — only share your thoughts on what the best way to systematize this is.

Ah! This explains why I couldn't get it to work way I expected whilst following definitions! No rush at all on this, I'll look at getting this merged as a string and then we can take from there.

**Comment by ItsJonQ** [comment]
> I wouldn't ask you to work on this — only share your thoughts on what the best way to systematize this is.

(Forgive me for stating obvious things. I'm just walking through all the steps)

Based on the original screenshot, it looks like there's a height difference between the controls (input x button).

---

#### ⚛️ Systematic Control UI

From a **systematic** perspective, all control elements should have the same height value. They should also have a series of height variants (e.g. small -> large).

In the case of G2 Components, the cluster of input + button would look like this:
```jsx
<Grid templateColumns="200px 1fr">
    <TextInput {...props} />
    <Button {...props} />
</Grid>
```

The sizes can be made larger through the `size` prop:

```jsx
<Grid templateColumns="200px 1fr">
    <TextInput size="large" {...props} />
    <Button size="large" {...props} />
</Grid>
```

The `Grid` component will put these components in place. Their widths can be specified through the `templateColumns` prop, which maps to the CSS `grid-template-columns` property for CSS grid.

---

#### 🛰 Spacing

You may notice there's nothing specified for spacing 🤔.

That's because `Grid` (and other multiple element layout components in G2) does this **automatically**.

For `Grid`, the space (or gap) between each element can be adjusted through the `gap` prop.

---

#### ✨ Examples

I've prepared an example CodeSandbox with various examples.

https://codesandbox.io/s/g2-control-cluster-size-example-4blni

Hope this helps!

**Pull request merged**